src/github_runner_manager/metrics/runner.py

```diff
@@ -155,9 +157,78 @@ def issue_events(
     Returns:
         A set of issued events.
     """
+    issued_events: set[Type[metric_events.Event]] = set()
+
+    if runner_metrics.installation_start_timestamp:
+        if not _issue_runner_installed(runner_metrics=runner_metrics, flavor=flavor):
+            # return to not issuing Runner{Start,Stop} metrics if RunnerInstalled metric
+            # could not be issued.
+            return issued_events
+        issued_events.add(metric_events.RunnerInstalled)
+
+    if runner_metrics.pre_job and _issue_runner_start(
+        runner_metrics=runner_metrics, flavor=flavor, job_metrics=job_metrics
+    ):
+        issued_events.add(metric_events.RunnerStart)
+    else:
+        # Return to not issuing RunnerStop metrics if RunnerStart metric could not be issued.
+        return issued_events
+
+    if runner_metrics.post_job and _issue_runner_stop(
+        runner_metrics=runner_metrics, flavor=flavor, job_metrics=job_metrics
+    ):
+        issued_events.add(metric_events.RunnerStop)
+
+    return issued_events
```

Comment:
```python
issued_events = {}

try:
    issued_events.update(runner_installed())
    issued_events.update(runner_start())
    issued_events.update(runner_stop())
except ...:
    pass
return issued_events
```

Reply:
I have updated the code as suggested, although the except clause now contains logging logic to improve debugging.